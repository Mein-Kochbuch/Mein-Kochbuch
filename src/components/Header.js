import React from "react";
import {StyleSheet, Text, View} from "react-native";

export default function Header({title}) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
