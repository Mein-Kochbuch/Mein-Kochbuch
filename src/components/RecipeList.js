import React from "react";
import {View} from "react-native";
import RecipeListItem from "./RecipeListItem";

export default function RecipeList({recipes, filter}) {

    return (
        <View>
            {recipes.map(recipe => <RecipeListItem key={recipe.pk} recipe={recipe}/>)}
        </View>
    )
}
