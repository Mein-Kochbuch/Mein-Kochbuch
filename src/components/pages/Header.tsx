import React from "react";
import {StyleSheet, Text, View} from "react-native";

interface HeaderProps {
    title: string
}

export default function Header({title}: HeaderProps) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
