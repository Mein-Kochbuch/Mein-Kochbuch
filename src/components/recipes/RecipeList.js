import React from "react";
import {View} from "react-native";
import RecipeListItem from "./RecipeListItem";

export default function RecipeList({recipes}) {

    return (
        <View>
            {recipes.map(recipe => <RecipeListItem key={recipe.pk} recipe={recipe}/>)}
        </View>
    )
}
