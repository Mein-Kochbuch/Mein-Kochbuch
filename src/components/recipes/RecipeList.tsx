import React from 'react';
import {FlatList} from 'react-native';
import RecipeListItem from './RecipeListItem';
import styled from 'styled-components/native';
import {RecipePreview} from '../../models/RecipePreview';

interface RecipeListProps {
  recipes: RecipePreview[];
  loadMoreRecipes: () => void;
}

export default function RecipeList({
  recipes,
  loadMoreRecipes,
}: RecipeListProps) {
  return (
    <ViewStyled>
      <FlatList
        data={recipes}
        renderItem={item => <RecipeListItem item={item} />}
        keyExtractor={item => item.id}
        onEndReached={loadMoreRecipes}
      />
    </ViewStyled>
  );
}

const ViewStyled = styled.View`
  height: 88%;
`;
