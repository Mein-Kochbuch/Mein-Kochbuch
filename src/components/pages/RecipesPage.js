import React from "react";
import Header from "./Header";
import styled from "styled-components/native";
import RecipeList from "../../components/recipes/RecipeList";

export default function RecipesPage({recipes, loadNext, title}) {

    const loadMoreRecipes = () => {
        loadNext()
    }

    return (
        <ViewStyled>
            <Header title={title}/>
            <RecipeList recipes={recipes} loadMoreRecipes={loadMoreRecipes}/>
        </ViewStyled>
    )

}

const ViewStyled = styled.View`
  height: 100%;
`
