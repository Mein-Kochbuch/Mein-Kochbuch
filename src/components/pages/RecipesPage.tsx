import React from 'react';
import Header from './Header';
import styled from 'styled-components/native';
import RecipeList from '../recipes/RecipeList';
import {RecipePreview} from '../../models/RecipePreview';

interface RecipePageProps {
  recipes: RecipePreview[];
  loadNext: () => void;
  title: string;
}

export default function RecipesPage({
  recipes,
  loadNext,
  title,
}: RecipePageProps) {
  const loadMoreRecipes = () => {
    loadNext();
  };

  return (
    <ViewStyled>
      <Header title={title} />
      <RecipeList recipes={recipes} loadMoreRecipes={loadMoreRecipes} />
    </ViewStyled>
  );
}

const ViewStyled = styled.View`
  height: 100%;
`;
