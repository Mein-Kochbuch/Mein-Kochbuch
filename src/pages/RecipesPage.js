import React from "react";
import Header from "../components/Header";
import styled from "styled-components/native";
import useRecipes from "../hooks/useRecipes";
import RecipeList from "../components/RecipeList";

export default function RecipesPage({title}) {
    const {recipes} = useRecipes()

    console.log(recipes[0])

    return (
        <ScrollViewStyled>
            <Header title={title}/>
            <RecipeList recipes={recipes}/>
        </ScrollViewStyled>
    );
}

const ScrollViewStyled = styled.ScrollView`
  height: 100%;
`
