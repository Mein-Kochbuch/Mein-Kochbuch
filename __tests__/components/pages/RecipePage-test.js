import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Header from "../../../src/components/pages/Header"
import RecipesPage from "../../../src/components/pages/RecipesPage";
import RecipeList from "../../../src/components/recipes/RecipeList";

const recipes = [{pk: "1", title: "test-title-1"}, {pk: "2", title: "test-title-2"}]

jest.mock("../../../src/hooks/useRecipePreviews", () => {
    return () => {
        return {recipes}
    }
});

jest.mock("../../../src/components/recipes/RecipeList", () => {
    return (props) => {
        return <div {...props}/>
    }
});

jest.mock("../../../src/components/list/Header", () => {
    return (props) => {
        return <div {...props}/>
    }
});


it('RecipeTest Test', () => {

    const loadNext = () => {
    }

    const component = renderer.create(
        <RecipesPage recipes={recipes} loadNext={loadNext} title={"test-title"}/>
    );

    expect(component.toJSON()).toMatchSnapshot()
    expect(component.toJSON().children.length).toBe(2)
    const testInstance = component.root
    expect(testInstance.findByType(Header).props.title).toBe("test-title")
    expect(testInstance.findByType(RecipeList).props.recipes).toStrictEqual(recipes)
    expect(testInstance.findByType(RecipeList).props.loadMoreRecipes).toBeTruthy()
})
