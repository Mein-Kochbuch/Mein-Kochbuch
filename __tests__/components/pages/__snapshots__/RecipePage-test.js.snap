// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTest Test 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
    ]
  }
>
  <div
    title="test-title"
  />
  <div
    loadMoreRecipes={[Function]}
    recipes={
      Array [
        Object {
          "pk": "1",
          "title": "test-title-1",
        },
        Object {
          "pk": "2",
          "title": "test-title-2",
        },
      ]
    }
  />
</View>
`;
